pkg/scripts/install.go

```diff
@@ -0,0 +1,173 @@
+package scripts
+
+import (
+	"net/url"
+	"path/filepath"
+
+	"github.com/golang/glog"
+
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/docker"
+	"github.com/openshift/source-to-image/pkg/errors"
+	"github.com/openshift/source-to-image/pkg/util"
+)
+
+// InstallResult structure describes the result of install operation
+type InstallResult struct {
+
```

Comment:
you can also put the Name here and then you don't need to return the map. You can just iterate over result and get the result based on the name.